tests/x11/gnucash.pm

```diff
@@ -32,16 +32,10 @@ sub run {
         send_key 'alt-c';
         assert_screen('test-gnucash-tips-closed');
     }
-    send_key 'ctrl-q';    # Exit
+    assert_and_click('gnucash-close-window');
+    assert_and_click('gnucash-close-without-saving-changes');
 
-    # arbitrary limit
-    for (1 .. 7) {
-        assert_screen [qw(test-gnucash-1 gnucash gnucash-save-changes generic-desktop)];
-        last              if match_has_tag 'generic-desktop';
-        send_key 'alt-c'  if match_has_tag 'test-gnucash-1';
-        send_key 'alt-w'  if match_has_tag 'gnucash-save-changes';
-        send_key 'ctrl-q' if match_has_tag 'gnucash';
-    }
+    assert_screen('generic-desktop');
```

Comment:
With this change I see two flaws. First you got rid of all alternatives. As I assume you only executed tests for a single product I am not sure if you broke something else instead, e.g. older products, please crosscheck that.

Second: Every test based on x11test should check for "generic-desktop" in the end anyway. Please check that tests now do not check twice which would be wasteful

Reply:
A best practice on test development is that each test start with a clean environment. How should I assert that gnucash is leaving a clean environment without checking the desktop? How do I assert that the window is closed?

It was doing the same before my changes. I only removed it from the if statement.